src/components/visualizations/chart/highcharts/plugins/adjustTickAmount.ts

```diff
@@ -125,32 +126,47 @@ export function adjustTicks(axis: IHighchartsAxisExtend): void {
 }
 
 /**
- * Align zero of secondary axis to primary axis
+ * Get axis score that increase 1 for data having positive and negative values
  * @param axis
+ * @return axis score, 1 means only negative or positive data, 2 means both
  */
-export function alignSecondaryAxis(axis: IHighchartsAxisExtend): void {
-    if (isPrimaryYAxis(axis)) {
-        // only handle on secondary axis
-        return;
+export function getYAxisScore(axis: IHighchartsAxisExtend): number {
+    let score: number = 0; // start with no data
+    const { dataMin, dataMax } = axis;
+    const yAxisMin = Math.min(0, dataMin);
+    const yAxisMax = Math.max(0, dataMax);
+
+    if (yAxisMin < 0) {
+        score += 1;
+    }
+    if (yAxisMax > 0) {
+        score += 1;
     }
+    return score;
+}
 
-    const {
-        chart: { axes },
-    } = axis;
-    const primaryAxis: IHighchartsAxisExtend = axes.find(
-        (axis: IHighchartsAxisExtend) => isYAxis(axis) && isPrimaryYAxis(axis),
-    );
-    if (!primaryAxis) {
-        return;
+/**
+ * Base on axis score which is bigger than another, will become base axis
+ * The other axis will be aligned to base axis
+ * @param yAxes
+ * @return array of Y axes, first is base axis, second is axis need aligned
+ */
+function getBaseYAxis(yAxes: IHighchartsAxisExtend[]): IHighchartsAxisExtend[] {
+    const [firstAxisScore, secondAxisScore] = yAxes.map(getYAxisScore);
+    if (firstAxisScore >= secondAxisScore) {
+        return [yAxes[0], yAxes[1]];
     }
+    return [yAxes[1], yAxes[0]];
+}
 
-    const { tickInterval } = axis;
+export function alignToBaseAxis(yAxis: IHighchartsAxisExtend, baseYAxis: IHighchartsAxisExtend): void {
+    const { tickInterval } = yAxis;
     for (
-        let direction: number = getDirection(primaryAxis, axis);
+        let direction: number = getDirection(baseYAxis, yAxis);
         direction !== ALIGNED;
-        direction = getDirection(primaryAxis, axis)
+        direction = getDirection(baseYAxis, yAxis)
```

Comment:
What do you think about replacing `for` with `while` loop?

Reply:
This can endup by an endless loop, if possible, please return `ALIGNED` by default for `getDirection` function

Reply:
I think `for` or `while` does not matter.
I have refactored `getDirection` and added unit test for it.
Please check :)
Thank for your good catch.